ucp/core.py

```diff
@@ -910,9 +910,9 @@ def init(options={}, env_takes_precedence=False, blocking_progress_mode=None):
             "in order to re-initate UCX with new options."
         )
     if env_takes_precedence:
-        for k in os.environ.keys():
-            if k in options:
-                del options[k]
+        options = options.copy()
+        for k, v in options.items():
+            options[k] = os.environ.get(f"UCX_{k}", v)
```

Comment:
Maybe we should add a warning or a `logger.debug` entry if both are set and have values?

Reply:
Not a warning, I think, because IME chatty warnings that always appear are swiftly ignored by end users.

Logging at debug level is reasonable. I think then it should occur in both branches (i.e. log whenever the environment and config overlap irrespective of which is shadowing).

Reply:
Done.